src/ui/game/player/complex_visual_player.cpp

```diff
@@ -260,13 +271,20 @@ void ComplexVisualPlayer::updateData(
         shouldBeVisible = (data.isVisible || settings.players.forceVisibility) && !isForciblyHidden;
     }
 
+    // Sets the badge to the right of the player name no matter the length :D
+    badgeWrapper->setPosition(playerName->getPosition() + CCPoint{parent->getAccountData().name.length() + 30.f * 1.20f, -25.f});
+    if (parent->getAccountData().name.length() >= 12) badgeWrapper->setPosition(playerName->getPosition() + CCPoint{parent->getAccountData().name.length() + 42.5f * 1.25f, -25.f});
+
```

Comment:
that's not how you're meant to use layouts at all. the entire reason you create a wrapper with a `RowLayout` (badgeWrapper) is so that you add both the name and the badge into that wrapper, and the layout will automatically place the badge along the name, without you needing to do any calculations yourself.